src/core/time.ts

```diff
@@ -435,11 +463,13 @@ export class Tag {
    * the given delay. The `microstep` of this time instant is ignored;
    * the returned time instant has a `microstep` of zero if the delay
    * is greater than zero. If the delay equals zero, the tag is returned
-   * unchanged with its current `microstep`.
+   * unchanged with its current `microstep`. If the delay is undefined,
+   * the tag is returned as it was.
    * @param delay The time interval to add to this time instant.
    */
-  getLaterTag(delay: TimeValue): Tag {
-    if (delay.isZero() || delay.isNever()) {
+  getLaterTag(delay: TimeValue | undefined): Tag {
+    if (delay === undefined || delay.isZero()) {
+      // FIXME: Should a tag + zero returns one microstep after the tag?
```

Comment:
Yes, here we definitely should add a microstep. The purpose of this method is to supply the runtime with a tag that is _later_ and thus needs to be strictly greater than the current one.

Reply:
Changing this method to add a microstep yields a lot of failures. My best guess is that we're adding a microstep elsewhere already...

Reply:
> Changing this method to add a microstep yields a lot of failures. My best guess is that we're adding a microstep elsewhere already...

Yes, we do: https://github.com/lf-lang/reactor-ts/blob/e8e6a4021b8a5a59decce758ccbb882706cdb6fd/src/core/action.ts#L136